core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java

```diff
@@ -244,4 +264,18 @@ public SqlStatements createCopy() {
     Collection<StatementCustomizer> getCustomizers() {
         return customizers;
     }
+
+    String preparedRender(String template, StatementContext ctx) {
+        try {
+            return Optional.ofNullable(
+                    templateCache.get(
+                            new AbstractMap.SimpleEntry<>(templateEngine, template),
+                            e -> e.getKey().parse(e.getValue(), ctx.getConfig())
+                                               .orElse(null)))
+                .orElse(cx -> templateEngine.render(template, cx))
+                .apply(ctx);
```

Comment:
This is difficult to understand at first glance. Could we add some comments or explanatory variables?